pkg/pushsync/pushsync.go

```diff
@@ -623,20 +503,25 @@ func (ps *PushSync) pushToNeighbour(ctx context.Context, peer swarm.Address, ch
 		}
 	}()
 
-	creditCtx, cancel := context.WithTimeout(context.Background(), time.Second)
-	defer cancel()
+	action, err := ps.prepareCredit(ctx, peer, ch, origin)
```

Comment:
`err` is re declared here from line 487, if error happens here will it be handled on line 499?